ssl/record/ssl3_record.c

```diff
@@ -361,7 +361,9 @@ int ssl3_get_record(SSL *s)
                     }
                 }
 
-                if (SSL_IS_TLS13(s) && s->enc_read_ctx != NULL) {
+                if (SSL_IS_TLS13(s)
+                        && s->enc_read_ctx != NULL
+                        && !BIO_get_ktls_recv(s->rbio)) {
                     if (thisrr->type != SSL3_RT_APPLICATION_DATA
                             && (thisrr->type != SSL3_RT_CHANGE_CIPHER_SPEC
                                 || !SSL_IS_FIRST_HANDSHAKE(s))
```

Comment:
GitHub's UI won't let me add this on the correct line, but I believe that for KTLS we should adjust the TLS1.3 specific check below on like 396 as for KTLS the extra byte for the record length has been stripped and the length should be validated against SSL3_RT_MAX_ENCRYPTED_LENGTH instead?